File: backend/stove-app-parent/src/main/java/by/shumpanov/stove/stove_app_parent/constructor/dto/ComponentDto.java
package by.shumpanov.stove.stove_app_parent.constructor.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.jpa.domain.AbstractPersistable;

import java.io.Serializable;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class ComponentDto implements Serializable {

    private Long id;
    @JsonProperty("name")
    private String name;
    @JsonProperty("description")
    private String description;
    @JsonProperty("is_required")
    private boolean isRequired;
    @JsonProperty("allow_multiple_choices")
    private boolean allowMultipleChoices;

    @JsonProperty("component_options")
    private List<ComponentOptionDto> options;
}

File: android/app/src/main/java/com/example/stove/data/remote/dto/ComponentDto.kt
package com.example.stove.data.remote.dto

import kotlinx.serialization.SerialName

data class ComponentDto(
    val id: Int,
    val name: String,
    val description: String,
    @SerialName("is_required") val isRequired: Boolean,
    @SerialName("allow_multiple_choices") val allowMultipleChoices: Boolean,
    @SerialName("component_options") val componentOptions: Boolean
)
